mw_context_vars_test.go

```diff
@@ -34,3 +37,37 @@ func TestContextVarsMiddleware(t *testing.T) {
 		{Path: "/test/path", Code: 200, BodyMatch: `"X-Request-Id":"`},
 	}...)
 }
+
+func TestMiddlewareContextVars_ProcessRequest_cookies(t *testing.T) {
+
+	req, _ := http.NewRequest(http.MethodGet, "/", nil)
+	res := httptest.NewRecorder()
+
+	req.Header.Set("Cookie", "abc=123; def=456")
+
+	err, code := (&MiddlewareContextVars{}).ProcessRequest(res, req, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if code != http.StatusOK {
+		t.Fatal(errors.New("non 200 status code"))
+	}
+
+	ctx := req.Context().Value(ContextData)
```

Comment:
I think using `ctxGetData(req)` would simplify code in this test and avoid type asserts in if statements below (hard to read for me)